datapipelines/src/test/java/io/customer/datapipelines/sdk/AndroidLifecyclePluginTests.kt

```diff
@@ -0,0 +1,296 @@
+import android.os.Bundle
+import androidx.test.platform.app.InstrumentationRegistry
+import com.segment.analytics.kotlin.android.plugins.AndroidLifecyclePlugin
+import com.segment.analytics.kotlin.core.Analytics
+import com.segment.analytics.kotlin.core.Configuration
+import com.segment.analytics.kotlin.core.Storage
+import com.segment.analytics.kotlin.core.TrackEvent
+import io.customer.datapipelines.config.DataPipelinesModuleConfig
+import io.customer.datapipelines.core.UnitTest
+import io.customer.datapipelines.extensions.updateAnalyticsConfig
+import io.customer.datapipelines.stubs.TestCoroutineConfiguration
+import io.customer.datapipelines.utils.TestRunPlugin
+import io.customer.datapipelines.utils.mockHTTPClient
+import io.customer.sdk.CustomerIOBuilder
+import io.customer.sdk.android.CustomerIO
+import io.customer.sdk.extensions.random
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.verify
+import java.time.Instant
+import java.util.Date
+import java.util.UUID
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.assertFalse
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.put
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE)
+class AndroidLifecyclePluginTests {
+    private val lifecyclePlugin = AndroidLifecyclePlugin()
+
+    private lateinit var analytics: Analytics
+    private val mockContext = spyk(InstrumentationRegistry.getInstrumentation().targetContext)
+    private val mockApplication = mockk<Application>(relaxed = true)
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val testDispatcher = UnconfinedTestDispatcher()
+    private val testScope = TestScope(testDispatcher)
+
+    init {
+        setupMocks()
+    }
+
+    private fun setupMocks() {
+        val packageInfo = PackageInfo().apply {
+            versionCode = 100
+            versionName = "1.0.0"
+        }
+
+        val packageManager = mockk<PackageManager> {
+            every { getPackageInfo("com.foo", 0) } returns packageInfo
+        }
+        every { mockApplication.packageName } returns "com.foo"
+        every { mockApplication.packageManager } returns packageManager
+
+        mockkStatic(Instant::class)
+        every { Instant.now() } returns Date(0).toInstant()
+        mockkStatic(UUID::class)
+        every { UUID.randomUUID().toString() } returns String.random
+
+        mockHTTPClient()
+    }
+
+    private fun createTestAnalyticsInstance(moduleConfig: DataPipelinesModuleConfig): Analytics {
```

Comment:
I think you are referring to methods in `Analytics.kt` I can do it, but it would make that file a lot heavier. 

I wanted to keep the `unit` test which don't need `application` instance separate with the `androidTest` which do need it. 

But I can combine the logic there if that helps.

Reply:
Resolving as the code is updated